lib/commands/exam.js

```diff
@@ -120,141 +124,98 @@ module.exports = TestCommand.extend({
 
   /**
    * Customizes the Testem config to have multiple test pages if attempting to
-   * run in parallel. It is important that the user specifies the number of
-   * launchers to run in parallel in their testem.json config.
+   * run in parallel or load-balance. It is important that the user specifies
+   * the number of launchers to run in parallel in their testem.json config.
    *
+   * @param {Object} commandOptions
    * @override
    */
   _generateCustomConfigs(commandOptions) {
     const config = this._super._generateCustomConfigs.apply(this, arguments);
 
-    if (this.validator.shouldParallelize) {
-      // Get the testPage from the generated config or the Testem config and
-      // use it as the baseUrl for the parallelized test pages
-      const baseUrl = config.testPage || this._getTestPage(commandOptions.configFile);
-      const count = commandOptions.split;
-      let partitions = commandOptions.partition;
-
-      if (!partitions) {
-        partitions = [];
-        for (let i = 0; i < commandOptions.split; i++) {
-          partitions.push(i + 1);
-        }
-      }
+    if (!commandOptions.loadBalance && !commandOptions.replayExecution && !commandOptions.parallel) return config;
 
-      if (Array.isArray(baseUrl)) {
-        const command = this;
-        config.testPage = baseUrl.reduce(function(testPages, baseUrl) {
-          return testPages.concat(command._generateTestPages(baseUrl, partitions, count));
-        }, []);
-      } else {
-        config.testPage = this._generateTestPages(baseUrl, partitions, count);
+    config.testPage = getMultipleTestPages(config, commandOptions);
+
+    if (commandOptions.loadBalance || commandOptions.replayExecution) {
+      config.custom_browser_socket_events = this._addLoadBalancingBrowserSocketEvents(config);
```

Comment:
Any reason why this is snake case? Do we get this option from elsewhere (Testem)?

Reply:
correct, this is an option that we get from Testem.